pgwire: log pgwire errors when writing output

otan opened the pull request:
Release note (general change): Added ability to further debug
connections shut down automatically by the server.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59460)
<!-- Reviewable:end -->

rafiss reviewed (changes requested):
awesome! interesting design pattern

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59460#-:-MS3_-Dv8hc1h0d9liw-:b-t0zqq5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/pgwire/conn.go, line 414 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59460#-MS3ZOOQ6rbJzCF6W_vi:-MS3ZOOQ6rbJzCF6W_vj:b-ccmgs8) ([raw file](https://github.com/cockroachdb/cockroach/blob/c554686854f7eb82d94af7586c9466c0329e5968/pkg/sql/pgwire/conn.go#L414)):*
> ```Go
> 					ctx, &c.readBuf, timeReceived, intSizer.GetUnqualifiedIntSize(),
> 				); err != nil {
> 					return false, err
> ```

i guess it doesn't matter so much, but we're doing `return false, <non-nil err>`. do you think it's more or less clear to return `true` here instead?
___
*[pkg/sql/pgwire/conn.go, line 456 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59460#-MS3Z3vK2DiP5yj2dWju:-MS3Z3vL-AYSic1dn9Ti:bmtssr6) ([raw file](https://github.com/cockroachdb/cockroach/blob/c554686854f7eb82d94af7586c9466c0329e5968/pkg/sql/pgwire/conn.go#L456)):*
> ```Go
> 				return false, nil
> 			default:
> 				return true, c.stmtBuf.Push(
> ```

previously we weren't breaking the loop in this case. was that a bug?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/sql/pgwire/conn.go, line 414 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59460#-MS3ZOOQ6rbJzCF6W_vi:-MS4FMXA62bsBq_IrDnm:b88m8go) ([raw file](https://github.com/cockroachdb/cockroach/blob/c554686854f7eb82d94af7586c9466c0329e5968/pkg/sql/pgwire/conn.go#L414)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i guess it doesn't matter so much, but we're doing `return false, <non-nil err>`. do you think it's more or less clear to return `true` here instead?
</blockquote></details>

generally for errors i like to return the 0 value, i think false makes sense here.
___
*[pkg/sql/pgwire/conn.go, line 456 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59460#-MS3Z3vK2DiP5yj2dWju:-MS4FSPGCEl0lKLXq3bl:b-4phy1c) ([raw file](https://github.com/cockroachdb/cockroach/blob/c554686854f7eb82d94af7586c9466c0329e5968/pkg/sql/pgwire/conn.go#L456)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

previously we weren't breaking the loop in this case. was that a bug?
</blockquote></details>

hmm, i'm going to revert to the old thing.
not sure, at least returning false means the user likely has a chance of seeing the error message


<!-- Sent from Reviewable.io -->

otan committed:
pgwire: log more pgwire errors when writing output

Release note (general change): Added ability to further debug
connections shut down automatically by the server.

otan commented:
thanks!

bors r=rafiss
